mne/preprocessing/_csd.py

```diff
@@ -0,0 +1,197 @@
+    n_legendre_terms : int
+        number of Legendre terms to evaluate (if g_matrix or h_matrix
+        is provided this will be ignored)
+    head : float
+        The head radius (unit sphere). Defaults to 1.
+    g_matrix : ndarray, shape (n_channels, n_channels) | None
+        The matrix oncluding the channel pairwise g function. If None,
+        the g_function will be computed from the data (default).
+    h_matrix : ndarray, shape (n_channels, n_channels) | None
+        The matrix oncluding the channel pairwise g function. If None,
+        the h_function will be computed from the data (default).
+    n_jobs : int
+        The number of processes to run in parallel. Note. Only used for
+        Epochs input. Defaults to 1.
+    copy : bool
+        Whether to overwrite instance data or create a copy.
+
+    Returns
+    -------
+    inst_csd : instance of Epochs or Evoked
+        The transformed data. Output type will match input type.
+    """
+
+    if not any([isinstance(inst, BaseRaw), isinstance(inst, BaseEpochs),
+                isinstance(inst, Evoked)]):
+        raise ValueError('An instance of Raw, Epochs or Evoked ' +
+                         'must be provided')
+
+    if 'csd' in inst.info['comps']:
+        raise ValueError('CSD already applied, should not be reapplied')
+
+    if copy is True:
+        out = inst.copy()
+    else:
+        out = inst
+
+    if picks is None:
+        picks = pick_types(inst.info, meg=False, eeg=True, exclude='bads')
+    elif not (isinstance(picks, list) or isinstance(picks, np.ndarray)):
+        raise ValueError('picks must be a np.ndarray or list, ' +
+                         ' got %s' % type(picks))
+
+    if len(picks) == 0:
+        raise ValueError('No EEG channels found.')
+
+    for name, matrix in dict(g_matrix=g_matrix, h_matrix=h_matrix).items():
+        if matrix is not None:
+            if (not isinstance(matrix, np.ndarray) or
+                    matrix.shape != (len(picks), len(picks))):
+                raise ValueError('An array that is shape n_channels x ' +
+                                 'n_channels is required for %s, got ' % name +
+                                 '%s %s' % (type(matrix), matrix.shape))
+
+    if lambda2 is None:
+        lambda2 = 1e-5
+
+    if (not (isinstance(lambda2, float) or isinstance(lambda2, int)) or
+            0 > lambda2 or lambda2 > 1):
+        raise ValueError('lambda2 must be a float less than 1, '
+                         'got %s' % lambda2)
+
+    if (not (isinstance(stiffness, float) or isinstance(stiffness, int)) or
+            stiffness < 0):
+        raise ValueError('stiffness must be non-negative numeric, '
+                         'got %s' % stiffness)
+
+    if not isinstance(n_legendre_terms, int) or n_legendre_terms < 1:
+        raise ValueError('n_legendre_terms must be an int greater than 0, '
+                         'got %s' % n_legendre_terms)
```

Comment:
_validate type (which will give TypeError) then a check for < 1 (raise your own ValueError)